pkg/dryrun/dryrun_utils.go

```diff
@@ -0,0 +1,273 @@
+)
+
+var ResetColor = "\033[0m"
+
+func successColor(input string) string {
+	// Green
+	return "\033[32m" + input + ResetColor
+}
+
+func errorColor(input string) string {
+	// Red
+	return "\033[31m" + input + ResetColor
+}
+
+func warningColor(input string) string {
+	// Yellow
+	return "\033[33m" + input + ResetColor
+}
+
+func compareStatus(cmd *cobra.Command,
+	inputStatus map[string]interface{},
+	resultStatus map[string]interface{},
+) {
+	cmd.Println("# Status compare:")
+
+	isStatusMatch := compareStatusObj(cmd, true, "", inputStatus, resultStatus)
+
+	if isStatusMatch {
+		cmd.Println(successColor("\033[1m Expected status matches the actual status \033[0m"))
+	} else {
+		cmd.Println(errorColor("\033[1m Expected status does not match the actual status \033[0m"))
+	}
+}
+
+// compareStatusObj compares the expected inputMap against the actual resultMap
+// and prints success or error messages for mismatches. It supports nested structures,
+// including maps and arrays, by recursively comparing their contents. The function
+// updates isStatusMatch based on whether all expected values match the actual values.
+func compareStatusObj(cmd *cobra.Command, isStatusMatch bool, parentPath string,
+	inputMap, resultMap map[string]interface{},
+) bool {
+	// Sort keys for consistency
+	keys := make([]string, 0, len(inputMap))
+	for k := range inputMap {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		actualValue, exist := resultMap[key]
+		value := inputMap[key]
+
+		if !exist {
+			cmd.Println(errorColor(fmt.Sprintf("Key '%s' not found in result at path '%s'", key, parentPath)))
+
+			isStatusMatch = false
+
+			continue
+		}
+
+		path := parentPath + "." + key
+
+		switch typedV := value.(type) {
+		case string, int, int64, float64, bool:
+			v := fmt.Sprintf("%v", typedV)
+
+			av := fmt.Sprintf("%v", actualValue)
+
+			if v != av {
+				errorMatchPrint(cmd, path, v, av)
+
+				isStatusMatch = false
+			} else {
+				successMatchPrint(cmd, path, v, av)
+			}
+
+		case []interface{}:
+			// Handle array of objects and non-object arrays
+			actualArray, ok := actualValue.([]interface{})
+			if !ok {
+				notArrayPrint(cmd, path, key)
+
+				isStatusMatch = false
+
+				continue
+			}
+
+			isStatusMatch = handleArrayInterface(cmd, typedV, actualArray, path, isStatusMatch)
+
+		case map[string]interface{}, map[interface{}]interface{}:
+			mapObj, ok := convertMapStringKey(typedV).(map[string]interface{})
+			if !ok {
+				failedParsePrint(cmd, path)
+
+				isStatusMatch = false
+
+				continue
+			}
+
+			mapActualObj, ok := convertMapStringKey(actualValue).(map[string]interface{})
+			if !ok {
+				failedTypeMissMatch(cmd, path, reflect.TypeOf(actualValue).String())
+
+				isStatusMatch = false
+
+				continue
+			}
+
+			// Recursive call for nested maps
+			isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, mapObj, mapActualObj)
```

Comment:
Previous discussion: https://github.com/open-cluster-management-io/config-policy-controller/pull/329#discussion_r1941892136

```suggestion
		case map[string]interface{}:
			mapObj := typedV.(map[string]interface{}) // guaranteed? Or maybe unnecessary?

			mapActualObj, ok := convertMapStringKey(actualValue).(map[string]interface{})
			if !ok {
				failedTypeMissMatch(cmd, path, reflect.TypeOf(actualValue).String())

				isStatusMatch = false

				continue
			}

			// Recursive call for nested maps
			isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, mapObj, mapActualObj)

		case map[interface{}]interface{}:
			failedParsePrint(cmd, path)
```

And actually, that new `case map[interface{}]interface{}:` might not be necessary at all, if it is handled well enough in the existing default case.

Reply:
```
case map[string]interface{}:
			mapObj := typedV.(map[string]interface{}) // guaranteed? Or maybe unnecessary?
```

guaranteed so unnecessary

Reply:
why unnecessary case is failedParsePrint? The reason  I addded map[interface{}]interface{} case because in testcase, parsing yaml turn to `map[interface{}]interface{}`. so I had to change to `map[string]interface{}` manually

Reply:
oh "yaml.UnmarshalStrict()" maybe resolve! the issue?

Reply:
Oh my god fixed Thanks Justin I use yaml.v3 then resolve everything!!